test/e2e/framework/helpers.go

```diff
@@ -168,6 +170,24 @@ func (f *Framework) MustGetPods(t *testing.T, namespace string) *v1.PodList {
 	return pods
 }
 
+func (f *Framework) IsFeatureGateEnabled(t *testing.T, name string) bool {
```

Comment:
suggestion: could we refactor and reuse the code at https://github.com/openshift/cluster-monitoring-operator/blob/0e876ff8f315847f3699af1ed9e8cbce3e34d6c3/pkg/operator/operator.go#L420? that would also allow testing it.

Reply:
That looks a bit complicated since it depends on informers but as part as this test is concerned IIUC checking feature gate status is good enough for us since we don't have a long running cluster in CI and we don't need to keep a informer to check the featuregate change for test.

Reply:
(Now that I think about it, I don't see any safeguard on our side making sure that the techpreview clusters really have metricsserver enabled, we could end up always running prom-adapter tests... Let's continue discussing this somewhere else :))